### PR TITLE
Twinmux emu cmssw925

### PR DESCRIPTION
**Pull request by gkaratha**
A new TwinMux  emulator. The results checked in 2017 data are here: https://gkaratha.web.cern.ch/gkaratha/bmtf/twinmux_comparisson/

### CODE REVIEW

File: L1Trigger/L1TwinMux/plugins/#L1TwinMuxProducer.cc# [hunk 1]
```diff
@@ -0,0 +1,94 @@
+//-------------------------------------------------
```
Comment (thomreis) [inline L1Trigger/L1TwinMux/plugins/#L1TwinMuxProducer.cc#:1]: What is this file?

**Review by thomreis — COMMENTED** [review]
I think the package should be L1Trigger/L1TTwinMux to be consistent with the other L1T components.

What happens to the old TwinMux emulator in L1Trigger/L1TMuonBarrel? I guess it should be removed from there. Does this not give a conflict with the same module name?

In general, with making a new package like this we loose all the history of the previous emulator. Perhaps doing it in two steps would be better: First change the files directly in the L1Trigger/L1TMuonBarrel version and then move them in a second step.

There is a .root file in the test directory and another binary. Can they be removed?

**Comment by thomreis** [comment]
Or, if that is easier, move the old twinmux to its own package first and then change it.

**Comment by gkaratha** [comment]
Hi Thomas,

I will remove the root file. No actually there are no modules with the same name. While we can first change the emulator in L1TMuonBarrel and then move it, it will still be a massive deletion of the old files since they are completely different now.

**Comment by rekovic** [comment]
@gkaratha
Thanks.

Please remove `L1Trigger/L1TwinMux/plugins/#L1TwinMuxProducer.cc#`
it looks like it is a duplicate of `L1Trigger/L1TwinMux/plugins/L1TwinMuxProducer.cc`

Please rename the package to L1TTwinMux.  `L1T` is used to difrerentiate Stage2 from Legacy (run1) code.  Also please make sure all the TwinMux needed code is within that directory.

**Comment by gkaratha** [comment]
@rekovic @thomreis Thanks for the comments. I will change names, delete the root and the "#"-file and make a (different) PR in steps so we have the history from old -> new. This will stay open for a while in order to have the code available.

**Comment by thomreis** [comment]
@gkaratha I think this can be closed.

**Comment by rekovic** [comment]
This is superseded by https://github.com/cms-l1t-offline/cmssw/pull/589

**Pull request closed without merging**